keps/core/model-cross-worksapce-entitlements.md

```diff
@@ -0,0 +1,1004 @@
+        "userInfo": {
+            "username": "admin",
+            "uid": "014fbff9a07c",
+            "groups:": [
+            - "system:authenticated",
+            - "some-custom-group"
+            ],
+            "extra": {
+                "some-key": [
+                - "some-value",
+                - "some-other-value"
+                ]
+            }
+        },
+        "requestInfo": {
+            "clusterName": "f5865fce",
+            "clusterPath": "root:management:us-west-invoices"
+        },
+        "entitlement": {
+            "clusterName": "33bab531",
+            "kind": "WidgetAccess",
+            "apiVersion": "acme.corp/v1",
+            "spec": {
+                "something": {
+                    "meaningful": "here"
+                }
+            }
+        }
+    },
+    "status": {
+        "entitled": true,
+        "disqualified": false,
+        "reason": "All users in `root:management:us-west-invoices` are entitled to `meaningful:here`."
+    }
+}
+```
+
+### Tenancy-Based Implementation
+
+In addition to providing the abstract concept of and endpoints for entitlement review, `kcp`
+will also provide a simple implementation tied to workspaces as tenancy boundaries. This
+implementation is intended to be sufficient for use-cases where the complexity of entitlements
+is low, the rate of change is sufficiently low and the incidence of strict entitlement rules
+is rare. This implementation will make use of two objects in the `entitlements.tenancy.kcp.io` group:
+
+```go
+type EntitlementPolicy struct {
+    metav1.TypeMeta `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // Entitlements holds all the entitlements that this EntitlementPolicy
+    // groups. 
+    // +optional
+    Entitlements []kcpcorev1alpha1.Entitlement `json:"entitlements"`
+}
+
+type EntitlementPolicyBinding struct {
+    metav1.TypeMeta `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // EntitlementPolicyRef is a reference to the entitlement policy
+    // that this binding applies to.
+    EntitlementPolicyRef EntitlementPolicyRef `json:"entitlementPolicyRef"`
+
+    // Children determines if this policy binding provides the entitlement just to the
+    // workspace in which it is created, or also to the child workspaces underneath that.
+    // Optional, defaults to false.
+    // +optional
+    Children boolean `json:"children,omitempty"`
+}
+
+type EntitlementPolicyRef struct {
+    // ClusterPath is the human-readable path to the logical cluster in which the entitlement
+    // policy is registered.
+    ClusterPath logicalcluster.Path `json:"clusterPath"`
+
+    // Name is the name of the entitlement policy.
+    Name string `json:"name"`
+}
+```
+
+A service provider with permissions to create `EntitlementPolicy` will do so using the
+normal `kcp` API:
+
+```
+> POST /clusters/<entitlement-provider-cluster>/apis/entitlements.tenancy.kcp.io/v1alpha1/entitlementpolicies
```

Comment:
as above, IMO the wrong direction. Put these into the entitled workspace. The current design here means that number of policies grow with number of entitled workspaces. E.g. this means for every home workspace you need this object too. IMO a data model that does not fit the scaling dimensions.

Reply:
Oh, further down:

> Entitlement policy bindings will be stored in tenant clusters

Maybe just a typo here? `s/<entitlement-provider-cluster>/<tenant-cluster>/` ?

Reply:
Yes! Also, the current design requires that the service (e.g. APIExport) and entitlement share a logical cluster.